velox/exec/Unnest.h

```diff
@@ -39,21 +39,46 @@ class Unnest : public Operator {
   bool isFinished() override;
 
  private:
-  // Generate output for 'size' input rows starting from 'start' input row.
-  //
-  // @param start First input row to include in the output.
-  // @param size Number of input rows to include in the output.
+  // Represents the range of rows to process and indicates that first and last
+  // row may need to be processed partially.
+  struct RowRange {
+    // First input row to be included in the output.
+    vector_size_t start;
+
+    // Number of input rows to be included in the output.
+    vector_size_t size;
+
+    // Processing of the first input row begins at index `firstRowStart_` and
+    // ends at 'firstRowEnd'.
+    vector_size_t firstRowEnd;
```

Comment:
@jinchengchenghh I believe we need only 4 numbers to describe the range of rows.

- start: top-level row where the range starts
- firstRowStart: where in the 'start' top-level row to start
- end:  top-level row where the range ends
- lastRowEnd: where in the 'end' top-level row to end

'end' can be replaced with size: = end - start

However, in the code I'm seeing 5 numbers:

- start
- size
- firstRowStart_
- firstRowEnd
- lastRowEnd

I believe we do not need firstRowEnd. What do you think? CC: @rui-mo 

<img width="489" alt="Screenshot 2024-10-30 at 10 39 38 AM" src="https://github.com/user-attachments/assets/c1c99b03-8dfe-4b7a-8735-e44d181c4ffd">

Reply:
Yes, I use `lastRowEnd` to replace `firstRowEnd`, check the row count is 1 to use rawMaxSizes[firstRow] or lastRowEnd